pandas/tests/frame/test_constructors.py

```diff
@@ -1011,9 +1024,13 @@ def __iter__(self):
         tm.assert_frame_equal(result, expected)
 
     def test_constructor_iterator(self):
+        result = DataFrame(iter(range(10)))
+        expected = DataFrame(list(range(10)))
+        tm.assert_frame_equal(result, expected)
 
+    def test_constructor_list_of_iterators(self):
+        result = DataFrame([iter(range(10)), iter(range(10))])
         expected = DataFrame([list(range(10)), list(range(10))])
-        result = DataFrame([range(10), range(10)])
```

Comment:
A plain ``range`` isn't a iterator, so this fixes up the nomenclature.

The range case is now placed in ``test_constructor_list_of_ranges``, which is clearer.